src/lib/EDSL.mli

```diff
@@ -23,31 +37,76 @@ val nop : unit t
 val if_then_else :
   bool t -> unit t -> unit t -> unit t
 val if_then : bool t -> unit t -> unit t
+
+val if_seq:
+  t:unit t list ->
+  ?e:unit t list ->
+  bool t ->
+  unit t
+(** [condition c ~t ~e] is an alternate API for {!if_then_else} (when
+    [?e] is provided) or {!if_then} (otherwise) that assumes “then”
+    and “else” bodies to be lists for {!seq} construct. *)
+
 val seq : unit t list -> unit t
 val not : bool t -> bool t
-val printf : ('a, unit, string, unit t) format4 -> 'a
-val file_exists : string -> bool t
+
+val file_exists : string t -> bool t
+
+(** {3 Switch Statements } *)
+
 val switch :
-  (bool t * unit t) list ->
-  default:unit t -> unit t
+  [ `Case of bool t * unit t | `Default of unit t ] list -> unit t
+(** Create a switch statement from a list of {!case} and optionally a
+    {!default} (the function raises an exception if there are more
+    than one default cases). *)
+
+val case :
+  bool t ->
+  unit t list ->
+  [> `Case of bool t * unit t ]
+(** Create a normal case for a {!switch} statement. *)
+
+val default : unit t list -> [> `Default of unit t ]
+(** Create the default case for a {!switch} statement. *)
+
+
+(**/**)
+val make_switch :
+  (bool Language.t * unit Language.t) list ->
+  default:unit Language.t -> unit Language.t
+(**/**)
+
+
 val write_output :
-  ?stdout:string ->
-  ?stderr:string ->
-  ?return_value:string -> unit t -> unit t
-val write_stdout : path:string -> unit t -> unit t
+  ?stdout:string t ->
+  ?stderr:string t ->
+  ?return_value:string t -> unit t -> unit t
+
+val write_stdout : path: string t -> unit t -> unit t
+
+(** {3 Literals } *)
 
 val string : string -> string t
 val int : int -> int t
-val bool : bool t
+val bool : bool -> bool t
+
 val output_as_string : unit t -> string t
 val feed : string:string t -> unit t -> unit t
 val ( >> ) : string t -> unit t -> unit t
 val loop_while : bool t -> body:unit t -> unit t
 
+type 'a cli_option = 'a Language.cli_option
+type 'a option_spec = 'a Language.option_spec
+type ('a, 'b) cli_options = ('a, 'b) Language.cli_options
```

Comment:
Can we give these type parameters more informative names (`'a` is the argument type?). 
And what are the instances where `'b` isn't `unit t` ?